app-office/libreoffice/patches/libreoffice-6.4.7.2.patchset

```diff
@@ -429,5 +429,876 @@ index 712a61d..2c3d5f1 100644
  ifeq ($(COM_IS_CLANG),TRUE)
 -- 
-2.26.0
+2.30.2
+
+
+From 7dee13bc3536e97794ffa0e226b326d26e7835da Mon Sep 17 00:00:00 2001
+From: Ken Mays <kmays2000@gmail.com>
+Date: Tue, 9 Feb 2022 12:15:02 +0300
+Subject: Add CSTDDEF include to O3TL
+
+
+diff --git a/include/o3tl/lru_map.hxx b/include/o3tl/lru_map.hxx
+index e822fde..33a1487 100644
+--- a/include/o3tl/lru_map.hxx
++++ b/include/o3tl/lru_map.hxx
+@@ -14,6 +14,7 @@
+ #include <cassert>
+ #include <list>
+ #include <unordered_map>
++#include <cstddef>
+ 
+ namespace o3tl
+ {
+-- 
+2.30.2
+
+
+From e6981799e2b90e34dff31190c163a10ee8e6d407 Mon Sep 17 00:00:00 2001
+From: Ken Mays <kmays2000@gmail.com>
+Date: Tue, 9 Feb 2022 12:26:19 +0300
+Subject: Fix Bison 3.8 issue with SQLbison.y
+
+
+diff --git a/connectivity/source/parse/sqlbison.y b/connectivity/source/parse/sqlbison.y
+index c9d423e..b444277 100644
+--- a/connectivity/source/parse/sqlbison.y
++++ b/connectivity/source/parse/sqlbison.y
+@@ -74,10 +74,15 @@ inline connectivity::OSQLInternalNode* newNode(const OUString& _newValue,
+ 
+ // yyi is the internal number of the rule that is currently being reduced
+ // This can be mapped to external rule number via the yyrmap.
++#if YYBISON >= 30800
++#define SQL_NEW_RULE 			newNode("", SQLNodeType::Rule, yyr1[yyrule])
++#define SQL_NEW_LISTRULE 		newNode("", SQLNodeType::ListRule, yyr1[yyrule])
++#define SQL_NEW_COMMALISTRULE   newNode("", SQLNodeType::CommaListRule, yyr1[yynrule])
++#else
+ #define SQL_NEW_RULE 			newNode("", SQLNodeType::Rule, yyr1[yyn])
+ #define SQL_NEW_LISTRULE 		newNode("", SQLNodeType::ListRule, yyr1[yyn])
+ #define SQL_NEW_COMMALISTRULE   newNode("", SQLNodeType::CommaListRule, yyr1[yyn])
+-
++#endif
+ 
+ extern connectivity::OSQLParser* xxx_pGLOBAL_SQLPARSER;
+ 
+-- 
+2.30.2
+
+
+From e942cd3ff9f320b1dc168d59179293b928a6bd3f Mon Sep 17 00:00:00 2001
+From: Ken Mays <kmays2000@gmail.com>
+Date: Tue, 9 Feb 2022 13:15:24 +0300
+Subject: Fix GCC11/12 issue with SAL_STREAM
+
+
+diff --git a/include/sal/log.hxx b/include/sal/log.hxx
+index 00d533a..1618222 100644
+--- a/include/sal/log.hxx
++++ b/include/sal/log.hxx
+@@ -161,13 +161,12 @@ inline char const * unwrapStream(SAL_UNUSED_PARAMETER StreamIgnore const &) {
+ 
+     @since LibreOffice 3.5
+ */
+-#if defined _LIBCPP_VERSION || (defined _MSC_VER && _MSC_VER >= 1915)
++#if defined _LIBCPP_VERSION || (defined _GLIBCXX_RELEASE && _GLIBCXX_RELEASE >= 12) \
++#if defined _LIBCPP_VERSION \
++	|| (defined _GLIBCXX_RELEASE >= 12 || (_GLIBCXX_RELEASE == 11 && __GLIBCXX__ > 20210428))) \
++	|| (defined _MSC_VER && _MSC_VER >= 1915)
+ #define SAL_STREAM(stream) \
+-    (::std::ostringstream() << stream).str()
+-#else
+-#define SAL_STREAM(stream) \
+-    (dynamic_cast< ::std::ostringstream & >(::std::ostringstream() << stream).str())
+-#endif
++	(::std::ostringstream() << stream).str()
+ 
+ /**
+     @page sal_log Basic logging functionality.
+-- 
+2.30.2
+
+
+From 35e272a5c21f7b1b36b58b3693b5d880cb22a6ee Mon Sep 17 00:00:00 2001
```

Comment:
Is this needed? (already included in the orcus patchset)

Reply:
Fixed